### DISCUSSION TITLE
Command line tools for SADL?

### DISCUSSION THREAD

**Discussion by Robert-Adelard** [ge-high-assurance/RACK#465] [Q&A] [answered]
Are there any command line tools for SADL? 

How are the SADL ontology files translated into OWL / RDF files for uploading into the RACK?

**Comment by tuxji** [comment]
As far as I know, everyone has been installing SADL into their Eclipse and using SADL in their Eclipse UI to translate SADL ontology files into OWL files for uploading to RACK.

SADL has a command line application which is undocumented and difficult to use - see [com.ge.research.sadl.applications](https://github.com/SemanticApplicationDesignLanguage/sadl/tree/development/sadl3/com.ge.research.sadl.parent/com.ge.research.sadl.applications).  You still need to install SADL into an existing Eclipse before you can run com.ge.research.sadl.applications from the command line.  I want to Dockerize an Eclipse installation on disk with SADL already installed and write an entry point script along with documentation to make the application easier to use.  I've done some preliminary work but it's been focused on SADL's logging.  I haven't started the Dockerization yet.

**Reply by Robert-Adelard** [reply]
Thanks - how does your "Dockerized Eclipse with SADL installed" relate to Web SADL?

**Reply by tuxji** [reply]
(I was on vacation, hence the delay.)  

I'm not too familiar with Web SADL.  What I know is that Web SADL is a minimal JavaScript-based and web-based IDE developed by TypeFox (https://www.typefox.io/theia/) which runs in your browser and communicates via REST with a SADL "language server".  I don't know how fully functional the combination of the Eclipse Theia IDE and the SADL language server is.  Abha may have a better idea than me whether Web SADL can translate SADL ontology files to OWL files for uploading to RACK, or if it's just an IDE for editing SADL files within your browser.  I think the real Eclipse desktop IDE with the SADL feature is still the canonical way to translate SADL ontology files to OWL files.

**Reply by Robert-Adelard** [reply]
Thanks - I'll await a response from @AbhaMoitra

**Comment by AbhaMoitra** [comment]
We are not using any command line tools to translate SADL to OWL (we generate OWL in eclipse IDE).

**Comment by tuxji** [comment, marked as answer]
We now have a dockerized SADL-Eclipse image (see <https://hub.docker.com/repository/docker/sadl/sadl-eclipse/>) which can translate a SADL project's SADL files to OWL files on the command line without needing a full install of Eclipse.  You are welcome to use this image and provide feedback. Note that SADL or Jena (I don't know which) uses an unpredictable order when writing triples to OWL files, so the diff command would not be useful for comparing old versus new OWL files (the diff command would find differences every time).
